src/main/java/edu/kit/compiler/data/AstVisitor.java

```diff
@@ -20,6 +20,7 @@
     default T visit(BlockStatementNode blockStatementNode) { throw new UnsupportedOperationException(); }
     default T visit(LocalVariableDeclarationStatementNode localVariableDeclarationStatementNode) { throw new UnsupportedOperationException(); }
     default T visit(IfStatementNode ifStatementNode) { throw new UnsupportedOperationException(); }
+    default T visit(StatementNode statementNode) { throw new UnsupportedOperationException(); }
```

Comment:
Why did you add this overload? It seems counterproductive in the visitor pattern.